### DISCUSSION TITLE
Add compatibility for terraform 1.8 and provider-defined functions

### DISCUSSION THREAD

**Discussion by fop-platformteam** [renovatebot/renovate#31282] [Suggest an Idea]
### Tell us more.

I would like to allow renovate to support terraform 1.8 with the provider-defined function:
### What would you like help with?

I would like a new feature to be developed

### Please tell us more about your question or problem

I have a module that simply deploys an helm chart to a cluster using the "helm_release" resource.
The configuration looks like the following:

```
resource "helm_release" "gitlab-runner_ec2" {
  namespace     = "my-namespace"
  name          = "my-name"
  repository    = "https://charts.gitlab.io"
  chart         = "gitlab-runner"
  version       = "0.68.1"
  values = [
    yamlencode({
      runners = {
        config = provider::toml::encode(
          {
            runners = [
              {
                cache = {
                  Path   = "my-cache"
                  Shared = true
                  Type   = "s3"
                  s3 = {
                    BucketLocation = var.aws_default_region
                    BucketName     = aws_s3_bucket.s3-cache.id
                    ServerAddress  = "s3.amazonaws.com"
                  }
                }
             }
            ]
          }
        )
        tags = each.value.tag_name
      }
    })
  ]
}
```
When I run renovate in the folder to detect the packages I receive the following:

```
...
DEBUG: Matched 3 file(s) for manager terraform: modules/gitlab-runner/main.tf, modules/gitlab-runner/outputs.tf, modules/gitlab-runner/variable.tf (repository=local)
DEBUG: failed to parse HCL file (repository=local, packageFile=modules/gitlab-runner/main.tf)
DEBUG: modules/gitlab-runner/outputs.tf has no content (repository=local)
DEBUG: manager extract durations (ms) (repository=local)
       "managers": {"terraform": 36}
DEBUG: Found 0 package file(s) (repository=local)
...
```

Basically the renovate bot fails to parse the HCL file due to the `provider::toml::encode` provider-defined function and therefore can't detect the helm_release chart and version. 

Commenting out the provider `provider::toml::encode` section fix the issue conferming the ipothesis

The provider-defined function has been introduced in version terraform 1.8, but I wasn't able to find anything on the renovate documentation stating that that version is not supported. 
I would like renovate bot to be able to support provider-defined functions of terraform 1.8

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment]
we use a library to parse HCL, so that needs an update. please report upstream.

https://www.npmjs.com/package/@cdktf/hcl2json?activeTab=readme
